mkdumprd

```diff
@@ -414,4 +417,24 @@ if ! is_fadump_capable; then
 	fi
 fi
 
+ro_regex="[[:space:]]ro[[:space:]]"
+status_target=$(get_target_from_path $(dirname "$VMCORE_CREATION_STATUS"))
+new_mntpoint=$(echo /vmcorestatus/$(get_mntpoint_from_target "$status_target") \
+		| tr -s "/")
+
+# For fadump, the kernel cmdline is the same for 1st and 2nd kernel. The
+# sysroot.mount unit will be generated based on 'root=' kernel cmdline, and
+# its mount r/w option will be taken from cmdline too. If we are mounting
+# the same device on both /sysroot and /vmcorestatus, we need to take the same
+# r/w option, otherwise the systemd mounting will fail. So we copy sysroot's
+# r/w option here, and remount it as writeable during kdump capture service
+# when needed.
+if is_fadump_capable && \
+   [[ $(get_root_fs_device) == "$status_target" ]] && \
+   [[ $(cat /proc/cmdline) =~ $ro_regex ]]; then
+        add_mount "$status_target" "" "ro " "$new_mntpoint"
```

Comment:
Does adding the `"ro"` here have any impact? In `to_mount` the `ro` should be converted back to `rw` anyway.

Reply:
No it won't, you see it is "ro ", not "ro", there is one extra space within, so it won't be converted to "rw" later in to_mount(). Personally I don't like the "hacking" here, but didn't think of a better approach...

Reply:
Arghh, ok I missed the extra space... That is really hacky...

Can't we use an different approach instead? If I understand the comment above correctly this hack is only needed when we dump to the rootfs. So we could only add the new mount point when it's _not_ a dump to rootfs and then check in dracut-kdump.sh whether the new mount point exists or not. If it exists we write the status there otherwise it must be a dump to rootfs and we can simply add the status to the target.
This should work, shouldn't it?

Reply:
OK, thanks for the suggestion, I will draft the code and see it can work as expected.